File: app/src/main/java/com/gmail/cristiandeives/motofretado/UpdateLocationBroadcastReceiver.java
package com.gmail.cristiandeives.motofretado;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.text.TextUtils;
import android.util.Log;

public class UpdateLocationBroadcastReceiver extends BroadcastReceiver {
    public static final String STOP_SERVICE_ACTION =
            UpdateLocationBroadcastReceiver.class.getPackage().getName()
                    + ".STOP_LOCATION_SERVICE_ACTION";

    private static final String TAG = UpdateLocationBroadcastReceiver.class.getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.v(TAG, "> onReceive([Context], [Intent])");

        if (TextUtils.equals(intent.getAction(), STOP_SERVICE_ACTION)) {
            Intent serviceIntent = new Intent(context, UpdateLocationService.class);
            context.stopService(serviceIntent);
        } else {
            Log.w(TAG, "I don't know what to do with the action " + intent.getAction());
        }

        Log.v(TAG, "< onReceive([Context], [Intent])");
    }
}

File: app/src/main/java/com/gmail/cristiandeives/motofretado/UpdateLocationBroadcastReceiver.kt
package com.gmail.cristiandeives.motofretado

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.support.annotation.MainThread
import android.util.Log

@MainThread
internal class UpdateLocationBroadcastReceiver : BroadcastReceiver() {
    companion object {
        internal val STOP_SERVICE_ACTION = "${UpdateLocationBroadcastReceiver::class.java.`package`.name}.STOP_LOCATION_SERVICE_ACTION"

        private val TAG = UpdateLocationBroadcastReceiver::class.java.simpleName
    }

    override fun onReceive(context: Context, intent: Intent) {
        Log.v(TAG, "> onReceive(context=$context, intent=$intent)")

        if (intent.action == STOP_SERVICE_ACTION) {
            val serviceIntent = Intent(context, UpdateLocationService::class.java)
            context.stopService(serviceIntent)
        } else {
            Log.w(TAG, "I don't know what to do with the action ${intent.action}")
        }

        Log.v(TAG, "< onReceive(context=$context, intent=$intent)")
    }
}